src/project.coffee

```diff
@@ -39,6 +39,14 @@ class Project extends Model
     @emitter = new Emitter
     @buffers ?= []
 
+    @directoryProviders = [new DefaultDirectoryProvider()]
+    atom.packages.serviceHub.consume(
+      'atom.directory-provider',
+      '^0.1.0',
+      # New providers are added to the front of @directoryProviders because
+      # DefaultDirectoryProvider is a catch-all that will always provide a Directory.
+      (provider) => @directoryProviders.splice(0, 0, provider))
```

Comment:
My gut says it's better to let the provider perform ad-hoc logic to decide if it can open the URI or not. For example, our default directory provider can work with `/the/path` or `file://the/path`.